ansible_base/activitystream/models/entry.py

```diff
@@ -42,6 +45,44 @@ class Meta:
     def __str__(self):
         return f'[{self.created}] {self.get_operation_display()} by {self.created_by}: {self.content_type} {self.object_id}'
 
+    @functools.cached_property
+    def changed_fk_fields(self):
+        """
+        :return: A dictionary of {field_name: pk} for any ForeignKey fields that have changed.
+            the pk is the new pk value, for changed fields.
+        """
+        changed_fks = {}
+
+        if self.changes is None:
+            return changed_fks
+
+        for op in ('added_fields', 'changed_fields', 'removed_fields'):
+            for field_name, value in self.changes.get(op, {}).items():
+                field = self.content_type.model_class()._meta.get_field(field_name)
```

Comment:
Since this is activity stream and there could, maybe me models that were delete during a migration but still exist in activity stream do we need a try catach around this incase something "goes missing"?

Reply:
This is a good call, but this is already a broken corner case in devel.